src/cares_wrap.cc

```diff
@@ -318,7 +318,7 @@ void cares_wrap_hostent_cpy(struct hostent* dest, struct hostent* src) {
 
   /* copy `h_name` */
   size_t name_size = strlen(src->h_name) + 1;
-  dest->h_name = node::Malloc<char>(name_size);
+  dest->h_name = reinterpret_cast<char*>(node::Malloc(name_size));
```

Comment:
@addaleax https://github.com/nodejs/node/pull/8482 it shows `dont-land-on-v6.x` and `v4.x`. May I still make a backport for it?